### PR TITLE
Current API client

### PR DESCRIPTION
**Pull request by abohlman**
Started adding database stuff

### CODE REVIEW

File: unicode/eval_engine/services.py [hunk 1]
```diff
@@ -17,25 +17,31 @@
 }
 
 
+# Makes the post request to the API and returns an object containing the API output
 def eval_engine(source, lang, test_in, test_out):
-    url = 'http://159.89.93.145:3000/submissions/?base64_encoded=false&wait=true'
+    url = 'http://159.89.93.145:3000/submissions/?base64_encoded=false&wait=true'   # wait=true makes the API call synchronous, halving the number of requests needed
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:22]: Can you put this url in the settings file?

File: unicode/eval_engine/services.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Makes the post request to the API and returns an object containing the API output
+def eval_engine(source, lang, test_in, test_out):
+    url = 'http://159.89.93.145:3000/submissions/?base64_encoded=false&wait=true'   # wait=true makes the API call synchronous, halving the number of requests needed
+    params = {
+        'source_code': source,
+        'language_id': lang,
+        'stdin': test_in,
+        'expected_output': test_out
+    }
+    r = requests.post(url, params=params)   # Sends the post request to the api
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:29]:
```suggestion
    return requests.post(url, params=params).json()   # Sends the post request to the api
```

File: unicode/eval_engine/services.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    eval_output = r.json()
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:30]:
```suggestion
```

File: unicode/eval_engine/services.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    return eval_output
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:31]:
```suggestion
```

File: unicode/eval_engine/services.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+
+# Function called by view.py in the editor. Submission_id is the primary key of the UserSubmission table
+def eval_setup(submission_id):
+    sub = UserSubmission.objects.get(id = submission_id)    # Gets an objects of all the attributes of the submission_id in UserSubmission
+    source = sub.submission
+    language = sub.language
+    # user = sub.submitter
+    test_case = ProblemTestCase.objects.filter(problem = sub.problem)
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:40]: You should change this to plural `test_cases` to show that its a list.

File: unicode/eval_engine/services.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    for case in test_case:
+        eval_object = eval_engine(source, lang[language], case.test_input, case.test_output)
+
+    if 'Error' in eval_object['status']['description']:
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:45]: Don't you want this if/else to be inside of the for loops? If not the control statement will only evaluate the last test case's eval_object.

File: unicode/eval_engine/settings.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (grunklejp) [inline unicode/eval_engine/settings.py:1]: We should be able to delete this file, correct?

File: unicode/eval_engine/services.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+# Judge0 assigns an integer to each supported language
+lang = {
+    'C': 4,
+    'C++': 10,
+    'C#': 16,
+    'Clojure': 18,
+    'Java': 26,
+    'JavaScript': 29,
+    'Octave': 32,
```
Comment (grunklejp) [inline unicode/eval_engine/services.py:14]: Never head of Octave.  Probably want to remove this, as it most likely won't get support on the client side.

File: unicode/eval_engine/result.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
+import json
```
Comment (grunklejp) [inline unicode/eval_engine/result.py:1]: Can we delete this file?

**Review by grunklejp — CHANGES_REQUESTED** [review]
Everything looks good, just some minor 🧽cleanup needed.

We also might want to think about writing some sort of boilerplate function to smush all of the test cases together or see if the API service can handle multiple test cases. Not doing this means we'll be making a network call for every test case rather than one network call per user submission.

Look into setting up some sort of authentication for the API like a secret 🔑
There are tons of bots that scan github for unauthenticated api endpoints to bombard, causing lots of damage or at the very least a pricey hosting bill. 

I also suggest that you make a .md file in the docs folder that goes through the process of setting up the Judge0 service and demonstrate making simple requests to it. Think of it as a very abridged version of the Judge0 docs, of course link to the official docs for further reading. 
 
------
This PR can be merge before creating the setup doc but if i'm not mistaken this still needs to build a dict of the graded test cases to pass back to editor/views right?

File: unicode/eval_engine/services.py (hunk 1 above)
Comment (kailynw) [reply unicode/eval_engine/services.py:22]: K

File: unicode/eval_engine/services.py (hunk 8 above)
Comment (kailynw) [reply unicode/eval_engine/services.py:14]: Octave is matlab

**Comment by kailynw** [comment]
This pull request is actually underlapping mine.  I've already done most of the all the changes you did plus more. So I'm going to close it. Thx

**Comment by grunklejp** [comment]
If you submit a PR that contains commits from another PR it usually
combines them into the new PR.

File: unicode/eval_engine/services.py (hunk 6 above)
Comment (abohlman) [reply unicode/eval_engine/services.py:45]: Yeah, I forgot to indent everything after adding the loop. I'll do that today

**Pull request closed without merging**